### DISCUSSION TITLE
Can Use BTT Filament Sensor With E3V2-BLTouch-3x3-HS-v4.2.2-v1.3.5

### DISCUSSION THREAD

**Discussion by salymy10000** [Jyers/Marlin#1248] [Q&A] [answered]
I use BigTreeTech Filament sensor for ender 3V2 4.2.2. I set Runout Distance is 7-10-25 but when i print alway show Filament Runout. Why???? Please help me tks all

**Comment by SpeedoFST** [comment]
https://github.com/Jyers/Marlin/discussions/1215

**Reply by salymy10000** [reply]
tks But when i modify fw is not good like E3V2-BLTouch-3x3-HS-v4.2.2-v1.3.5 and i thing in the FW dont Uncomment #define FILAMENT_MOTION_SENSOR

**Reply by Cal-LF** [reply]
BTT sensor have to #define FILAMENT_MOTION_SENSOR.

**Reply by salymy10000** [reply]
![gh](https://user-images.githubusercontent.com/35592129/127748324-f5b5f4d3-5632-4b63-8095-c784882fc52d.png)
i modify here but a cant seen the advan option
![224078117_824087618250490_5146135034157477511_n](https://user-images.githubusercontent.com/35592129/127748363-9f372dc8-e499-44fb-853d-b5a049e38d9b.jpg)

**Reply by salymy10000** [reply]
![Screenshot_20210801-011812_Chrome.jpg](https://user-images.githubusercontent.com/35592129/127749066-4f38699c-8379-4647-a049-94f74cbbd3cd.jpg)
Ok tks you ao much

**Reply by Cal-LF** [reply]
@salymy10000 Not understanding what you are asking. 
The BTT motion sensor does not show up in the advance menu. Only filament runout and runout length. I use 10mm runout length. Reference #1215
Also may have to change another setting if BTT is not working. 
Change runout state in Config.h  1429 #define FIL_RUNOUT_STATE     LOW        // Pin state indicating that filament is NOT present.
Try change to HIGH

**Comment by salymy10000** [comment, marked as answer]
Ok Tks everyone so much.
 Need Download source code 1.3.5
 - In the BLTouch-3x3 folder select both Config.h & config_adv.h files, right click them and select copy.
 - Go to the Marlin section, right click and paste to replace the existing Config.h & config_adv.h files with the ones from the BLTouch-3x3 folder.
 
![1](https://user-images.githubusercontent.com/35592129/127758272-1ac36c7b-ef2f-41c4-8834-a93a434532e5.png)

 
[Compiling.Steps.txt](https://github.com/Jyers/Marlin/files/6912101/Compiling.Steps.txt)

and 

uncomment

#define FILAMENT_MOTION_SENSOR for smart sensor

If the Z homing faise 

https://user-images.githubusercontent.com/35592129/127758207-a8701d92-395a-4916-a0cb-99676c07770f.mp4

You need
disabled //#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN.

**Reply by Cal-LF** [reply]
Is your printer now printing ok?

**Reply by salymy10000** [reply]
![20210801_195651.jpg](https://user-images.githubusercontent.com/35592129/127771695-c73eacae-7d49-4a85-80f6-5f9e912ca94e.jpg)
Yes very nice 
This is product when i test stuck filament
![20210801_154154.jpg](https://user-images.githubusercontent.com/35592129/127771740-66020818-97ae-41df-8f62-4ed4ffee3907.jpg)
![20210801_154157.jpg](https://user-images.githubusercontent.com/35592129/127771746-6ec716e1-cad3-4ca1-8a3b-08f40a364003.jpg)
